### PR TITLE
Fix forced default experiment creation

### LINKED ISSUE #4351: [BUG] Deleting the 'Default' Experiment breaks MLflow initialization
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ probably ] Yes. I can contribute a fix for this bug independently.
- [ maybe ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### System information
- MLflow docker image based on 'python:3.9-slim-buster'
- MLflow installed from official pip \ pypi
- MLflow version 1.16.0
- MLflow backend store: AWS RDS MySQL 8.0.23
- MLflow artifact root: AWS S3 bucket
- Python version: 3.9
- command: mlflow server --backend-store-uri mysql+pymysql://root:${mlflow_rds_root_password}@${address}/${stack_env} --default-artifact-root "s3://${s3_bucket}/${s3_bucket_prefix}/${stack_env}/" --host 0.0.0.0:8080

### Describe the problem
After running MLflow for the first time, against an AWS RDS backend store,
a chaos engineer could request MLflow to delete the 'Default' experiment, which is created by the db migration mechanisms used by MLflow ( SQLalchemy migrations ).

after the default experiment (id 0) is deleted, MLflow continues to function fine, **until** the next service restart,
if the service is restarted, then it would fail to initialize successfully with these errors:

```
2021/05/12 17:20:12 ERROR mlflow.cli: Error initializing backend store
2021/05/12 17:20:12 ERROR mlflow.cli: (pymysql.err.IntegrityError) (1062, "Duplicate entry '0' for key 'experiments.PRIMARY'")
[SQL: INSERT INTO experiments (experiment_id, name, artifact_location, lifecycle_stage) VALUES (0, 'Default', '"s3://aura-data-mlflow/MLflow/dev/"/0', 'active');]
(Background on this error at: http://sqlalche.me/e/14/gkpj)
Traceback (most recent call last):
...
...
...
pymysql.err.IntegrityError: (1062, "Duplicate entry '0' for key 'experiments.PRIMARY'")
[SQL: INSERT INTO experiments (experiment_id, name, artifact_location, lifecycle_stage) VALUES (0, 'Default', '"s3://aura-data-mlflow/MLflow/dev/"/0', 'active');]
```

Clearly, the migration process attempts to insert a new default experiment at id 0, while it was already there, but removed due to user request to delete it..

### Code to reproduce issue
User action based, but I guess someone could also do it with:

```
#!/usr/bin/env python
import mlflow
TRACKING_URI = 'https://mlflow-dev.yourdomain.com/'

mlflow.set_tracking_uri(TRACKING_URI)
mlflow.delete_experiment('0')
# OR 
client = mlflow.tracking.MlflowClient(TRACKING_URI)
client.delete_experiment('0')
```

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ X ] `area/server-infra`: MLflow server, JavaScript dev server
- [ X ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ X ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ X ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry

### PR DESCRIPTION
**Pull request by asaf400**
## What changes are proposed in this pull request?
 
Fixes #4351 
## How is this patch tested?

Just followed CONTRIBUTING.rst to setup a local dev server, and confirmed that after the fix,
mlflow works, while before the fix mlflow doesn't work..

### Is this a user-facing change?

- [ X ] No. You can skip the rest of this section.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ X ] `area/server-infra`: MLflow server, JavaScript dev server
- [ X ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ X ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ X ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:
- [ X ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/tracking/sqlalchemy_store.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def __init__(self, db_uri, default_artifact_root):
         if is_local_uri(default_artifact_root):
             mkdir(local_file_uri_to_path(default_artifact_root))
 
-        if len(self.list_experiments()) == 0:
+        if len(self.list_experiments(view_type=ViewType.ALL)) == 0:
```
Comment (asaf400) [inline mlflow/store/tracking/sqlalchemy_store.py:128]: when check if mlflow should create the default experiment, 
it should take ALL experiments into account , not just Active experiments, as per default behaviour of `list_experiments` function

**Comment by asaf400** [comment]
@andrewmchen @harupy Review required for this very small footprint fix

**Comment by harupy** [comment]
@asaf400 Thanks for the fix! The proposed change looks good to me. Could you add a test to verify the fix? For example,

```python
# tests/store/tracking/test_sqlalchemy_store.py

def test_sqlalchemy_store_can_be_initialized_when_default_experiment_has_been_deleted(tmpdir):
    db_uri = "sqlite:///{}/mlflow.db".format(tmpdir.strpath)
    store = SqlAlchemyStore(db_uri, ARTIFACT_URI)
    store.delete_experiment("0")
    assert store.get_experiment("0").lifecycle_stage == entities.LifecycleStage.DELETED
    SqlAlchemyStore(db_uri, ARTIFACT_URI)
```

**Review by harupy — APPROVED** [review]

**Comment by harupy** [comment]
LGTM

**Comment by asaf400** [comment]
Just got the message, yeah I am glad you added the tests,
Thanks for merging

**Pull request merged**